### ISSUE TITLE
Add csrf_name, csrf_value to default list of possible CSRF tokens to avoid false positive

### ISSUE THREAD

**Issue by mig5** [zaproxy/zaproxy#6787]
I am running the zap2docker-weekly Docker image against a site that uses the PHP Slim framework's CSRF middleware as seen at https://github.com/slimphp/Slim-Csrf

I am getting the 'Absence of Anti-CSRF Tokens' finding in ZAP even though the site uses CSRF tokens.

The Slim CSRF middleware uses CSRF names/values that appear like this in the HTML:

```
<input id="csrf_name" type="hidden" name="csrf_name" value="csrfxxxxxxxxxxxxxxx">
<input id="csrf_value" type="hidden" name="csrf_value" value="xxxxxxxxxxxxxxxxxxxxxxxx">
```

I [understand](https://github.com/zaproxy/zaproxy/blob/6e09bb40ed3de3ca634fe24df6fff3eea2397712/zap/src/main/java/org/zaproxy/zap/extension/anticsrf/AntiCsrfParam.java#L50-L65) that the CSRF checker checks against a list of common CSRF attribute names.

Could the above csrf_name and csrf_value be added to that list so that this false positive no longer occurs? 

Thanks!

**Comment by kingthorin** [comment]
The built-in list isn’t meant to be exhaustive, you can add entries on the fly. With a hook and the weekly.

**Comment by mig5** [comment]
Thanks, that sounds intriguing, if a bit vague (also, the link to 'Scan hook' on https://www.zaproxy.org/docs/docker/about/  404's, but I found it in the end by resorting to Google). I'll look into that... so far, I've not found any examples of using a 'hook' to extend an existing rule.

**Comment by thc202** [comment]
The hook would be used to add the CSRF token to ZAP. I don't see a problem adding `csrf_name` as default though.

Broken link being fixed in zaproxy/zaproxy-website#553, thanks for reporting.

**Comment by kingthorin** [comment]
If people think it’s widespread enough to include that’s fine.

Thanks for the heads-up on the 404.
